types/index.d.ts

```diff
@@ -7,6 +7,7 @@ import IndexHints = require('./lib/index-hints');
 import Utils = require('./lib/utils');
 
 export * from './lib/sequelize';
+export { default } from './lib/sequelize';
```

Comment:
I wanted to call this out. Idk if it should have its own issue or whatnot, but I didn't expect much pushback so I'm throwing it in.

When using Sequelize with native ESM (`"type":"module"`) the **only** valid way to import sequelize is:
```
import Sequelize from 'sequelize';
```
With that object being the one defined here: https://github.com/sequelize/sequelize/blob/main/lib/sequelize.js#L1369

Test for yourself with
```
import('sequelize').then(m => sequelize = m);
sequelize.default === require('sequelize') // true
sequelize.where() // throws error
```

Any "named exports" are _only_ good for Typescript typings, which is still really valuable, but should be treated as types _only_.
e.g. something like this: https://github.com/sequelize/sequelize/blob/main/types/test/where.ts#L2 _will fail_ when actually executed

Supporting named exports would be great, but that is not actually the case yet. I didn't want to get into a refactor of a bunch of other types/tests, but worth noting that the current advertised usage is invalid for ESM (it _may_ work with Babel-ified code, i haven't tested)

Typescript was already grabbing the default export if [allowSyntheticDefaultImports=true](https://www.typescriptlang.org/tsconfig/#allowSyntheticDefaultImports) was enabled, but it is nice to do it explicitly.

My suggestion would be to convert all the non-default export statements in the `index.d.ts` file to `export type ...` to enforce this behavior. Then fix up tests to match.

That _may_ be a breaking change for some people (again idk if it currently works with Babel).

On second thought I probably will make another issue for this (not a blocker at all to this PR btw)

Reply:
Opened https://github.com/sequelize/sequelize/issues/13132

Resolving this "conversation" because its noisy and unrelated to the PR